src/main/webapp/2Dscript.js

```diff
@@ -0,0 +1,86 @@
+// These are gobal variables.
+let map;
+let pose;
+let runId;
+let dataType;
+
+fetchData();
+/**
+ * This function fetchs pose data from the RunInfo servlet,
+ * it is an asynchronous call requiring initMap() to be
+ * called after the data is fully loaded.
+ */
+function fetchData() {
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  runId = urlParams.get('id');
+  dataType = urlParams.get('dataType');
+  fetch('/getrun?id=' + runId + '&dataType=' + dataType)
+      .then((response) => response.json())
+      .then((data) => pose = data)
+      .then(() => initMap());
+}
+
+/**
+ * This function embeds the Google map interface within the html
+ * along with drawing the pose trajectory and adding a button link.
+ */
+function initMap() {
+  // This intitalizes the map.
+  map = new google.maps.Map(document.getElementById('map'), {
+    center: {lat: pose[0].lat, lng: pose[0].lng},
+    zoom: 18,
+  });
+  // This draws the pose trejectory.
```

Comment:
*trajectory

Reply:
also just an extension of jared comment: you could take the word "this" out of comments